rust-symcrypt/src/eckey.rs

```diff
@@ -0,0 +1,167 @@
+//! Friendly rust types for CurveTypes.
+
+use lazy_static::lazy_static;
+use crate::{errors::SymCryptError, symcrypt_init};
+use symcrypt_sys;
+
+/// [`CurveType`] provides an enum of the curve types that can be used when creating an [`EcKey`].
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub enum CurveType {
+    NistP256,
+    NistP384,
+    Curve25519,
+}
+
+/// [`EcKey`] is a wrapper around symcrypt_sys::PSYMCRYPT_ECKEY, this is to let rust handle the drop and subsequent free.
+/// EcKey must be allocated after [`EcCurve`] and free'd before EcCurve is free'd.
+///
+/// Allocation for EcKey is handled by SymCrypt via SymCryptEcKeyAllocate, and is subsequently stored on the stack, therefore pointer will
+/// not move and Box<> is not needed.
+pub struct EcKey {
+    inner: symcrypt_sys::PSYMCRYPT_ECKEY,
+    curve: &'static EcCurve,
+}
+
+/// [`EcCurve`] is a wrapper around symcrypt_sys::PSYMCRYPT_ECURVE, this is to let rust handle the drop and subsequent free.
+/// EcCurve must be allocated before [`EcKey`] is allocated, and dropped after EcKey is dropped.
+pub(crate) struct EcCurve(pub(crate) symcrypt_sys::PSYMCRYPT_ECURVE);
+
+/// Impl for [`EcKey`]
+///
+/// [`new()`] returns a new EcKey object that has the key and curve allocated. This key must be allocated after the [`EcCurve`] has been allocated
+/// and dropped after [`EcCurve`] has been dropped.
+///
+/// [`inner()`] is an accessor to the inner field of the EcKey struct. Reference is not needed here since we are working with a raw SymCrypt pointer.
+///
+/// [`curve()`] is an accessor to the curve field of the EcKey struct. Reference is used here since EcKey should still maintain ownership of the EcCurve.
+impl EcKey {
+    pub(crate) fn new(curve: CurveType) -> Result<Self, SymCryptError> {
+        let ec_curve = &EcCurve::new(curve);
```

Comment:
```suggestion
        let ec_curve = EcCurve::new(curve);
```

I don't think you need the `&` here.